ui/src/features/Editor/components/Canvas/components/Nodes/BatchNode/index.tsx

```diff
@@ -16,9 +18,54 @@ export const baseBatchNode = {
 
 const minSize = { width: 250, height: 150 };
 
-const BatchNode: React.FC<BatchNodeProps> = ({ data, selected }) => {
+const BatchNode: React.FC<BatchNodeProps> = ({ data, selected, id }) => {
   const [_width, _setWidth] = useState(data.width ?? initialSize.width);
   const [_height, _setHeight] = useState(data.height ?? initialSize.height);
+  const { getNodes, setNodes } = useReactFlow<Node>();
+  const { handleNodeDropInBatch } = useBatch();
+
+  const getChildNodesBoundary = useCallback(() => {
+    const nodes = getNodes();
+    const childNodes = nodes.filter((node) => node.parentId === id);
+
+    let maxX = 0;
+    let maxY = 0;
+
+    childNodes.forEach((node) => {
+      if (node.measured) {
+        const rightEdge = node.position.x + (node.measured?.width ?? 0);
+        const bottomEdge = node.position.y + (node.measured?.height ?? 0);
+
+        maxX = Math.max(maxX, rightEdge);
+        maxY = Math.max(maxY, bottomEdge);
+      }
+    });
+
+    return {
+      width: Math.max(minSize.width, maxX + 8),
+      height: Math.max(minSize.height, maxY + 8),
+    };
+  }, [getNodes, id]);
+
+  const handleOnEndResize = useCallback(() => {
+    const allNodes = getNodes();
+    let updatedNodes = allNodes;
+    const initialParentCount = allNodes.filter((node) => node.parentId).length;
+
+    const nonBatchNodes = allNodes.filter((node) => node.type !== "batch");
+    nonBatchNodes.forEach((node) => {
+      updatedNodes = handleNodeDropInBatch(node, updatedNodes);
+    });
+    const finalParentCount = updatedNodes.filter(
+      (node) => node.parentId,
+    ).length;
+
+    if (finalParentCount !== initialParentCount) {
+      setNodes(updatedNodes);
+    }
+  }, [getNodes, setNodes, handleNodeDropInBatch]);
+
+  const bounds = getChildNodesBoundary();
```

Comment:
Just to be really safe,

```suggestion
  const bounds = useMemo(() => getChildNodesBoundary(), [getChildNodesBoundary]);
```